crates/sage-cloudpath/src/util.rs

```diff
@@ -0,0 +1,84 @@
+use crate::read_and_execute;
+use sage_core::spectrum::RawSpectrum;
+use tokio::io::AsyncReadExt;
+
+#[derive(Debug)]
+pub enum Error {
```

Comment:
Unify `CloudPathError` and `MzML` error into a single error type? Maybe check out thiserror or similar?